main/ertp/ERTP-Guide/Overview.md

```diff
@@ -0,0 +1,276 @@
+
+## Promises
+
+Several ERTP methods are *asynchronous* and instead of immediately returning their expected value, return a *promise* for that value.
+
+JavaScript implements `Promise` objects, and recently added the two commands `async` and `await` to simplify working with them. For general, and extensive, information about JavaScript's implementation, see either:
+- [javascript.info](https://javascript.info/async)
+- [Mozilla's Developer Docs](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous)
+
+**tyg todo: Needs beefing up about how Promises, async, and await are
+  actually used in contracts**
+
+
+## Alleged Names
+
+ERTP implements [*object capabilities*](https://agoric.com/documentation/glossary/#object-capabilities). You can only use an object and issue commands to it if you have access to that object, not just its human-readable name or similar. For example, I might know (or make a good guess), that the mint that makes quatloos has the human-understandable name of 'quatloos-mint'. But unless I have the actual object that is a `mint` object associated with the `quatloos` `brand` object, I can't use it to create a million `quatloos` and bet
+them all on Captain Kirk to win his gladitorial match on Triskelion (see the[ Wikipedia entry](https://en.wikipedia.org/wiki/The_Gamesters_of_Triskelion)).
+
+**tyg todo: Needs beefing up on how allegedNames are used in contracts
+  vs. getting and using the real objects**
+
+## Creating and sending a fungible asset
+
+In ERTP, digital assets are created by a [`mint`](). You must have
+access to a `mint` object to create new assets of the `mint` object's
+associated `brand`. 
+
+For instance, let's say we want to create a new community
+currency called 'BaytownBucks'. You start by getting access to a `mint`
+object whose associated brand is 'BaytownBucks'.
+```js
+import { makeMint } from '@agoric/ertp';
+const baytownBucksMint = makeMint('BaytownBucks');
+```
+
+Now, use the `mint` to create 1000 new BaytownBucks.
+
+```js
+const { issuer, mint } = produceIssuer('BaytownBucks');
+const BB1000 = amountMath.make(1000);
+const newPayment = mint.mintPayment(BB1000);
+const BBpurse = issuer.makeEmptyPurse();
+const BB100Amount = BBpurse.deposit(newPayment);
+```
+
+The act of minting created 1000 BaytownBucks and stored them together in a
+`payment`. We then deposited the `payment` into a `purse` for BaytownBucks.
+
+Now, let's distribute the BaytownBucks to others. To send assets
+in ERTP, we withdraw a `payment` from a `purse`.
+
+```js
+const paymentForAlice = purse.withdraw(10, `alice's community money`);
+```
+
+Like our `purse`, this `payment` contains BaytownBucks, but unlike purses,
+payments are used to represent tokens in transit. A `payment` can be
+sent to someone else, a `purse` should never be sent to someone else.
+
+Now let's send the `payment` to Alice as a message:
+
+```js
+alice.receivePayment(paymentForAlice);
```

Comment:
We can actually do this now! I think we should explain how to use the board here and actually send a payment to another user. We might want to do this in a link that can be clicked through to.
Without mention of the board:

```suggestion
aliceDepositFacet.receive(paymentForAlice);
```
With mention of the board: 
```js
// Alice gives us the boardId to use for the transfer: aliceDepositFacetBoardId
const aliceDepositFacet = await E(board).getValue(aliceDepositFacetBoardId);
E(aliceDepositFacet).receive(payment);
// Alice got her payment automatically deposited in her purse!
```